### PR TITLE
Add documentation for validating models

### PR DESCRIPTION
**Pull request by mdellweg**
fixes #5927
https://pulp.plan.io/issues/5927

### CODE REVIEW

File: docs/plugins/plugin-writer/concepts/index.rst [hunk 1]
```diff
@@ -39,6 +39,23 @@ the box.
    subclassing/import-export
 
 
+.. _validating-models:
+
+Validating Models
+-----------------
+
+Pulp ensures valitidy of its database models by carefully crafted serializers.
```
Comment (daviddavis) [inline docs/plugins/plugin-writer/concepts/index.rst:47]: validity

**Comment by daviddavis** [comment]
This is great. Would it be possible to also include some docs about using serializers with the stages api?

File: docs/plugins/plugin-writer/concepts/index.rst (hunk 1 above)
Comment (mdellweg) [reply docs/plugins/plugin-writer/concepts/index.rst:47]:
```suggestion
Pulp ensures validity of its database models by carefully crafted serializers.
```

File: docs/plugins/plugin-writer/concepts/index.rst [hunk 2]
```diff
@@ -39,6 +39,39 @@ the box.
...
+
+To create a ``MyModel`` from a ``data`` dictionary, the ``MyModelSerializer`` can be used like:
+
+.. code-block:: python
+
+     serializer = MyModelSerializer(data=data)
+     serializer.is_valid(raise_exception=True)
+     instance = serializer.create(serializer.validated_data)
+
+In the stages pipeline, you want to instanciate the content units without saving them to database
```
Comment (daviddavis) [inline docs/plugins/plugin-writer/concepts/index.rst:58]: instantiate

File: docs/plugins/plugin-writer/concepts/index.rst [hunk 3]
```diff
@@ -39,6 +39,39 @@ the box.
...
+     # In MyPluginFirstStage::run
+     # <...>
+     serializer = MyModelSerializer(data=data)
+     serializer.is_valid(raise_exception=True)
+     d_content = DeclarativeContent(
+         content=MyModel(**serializer.validated_data),
+         d_artifacts=[d_artifact],
+     )
+     await self.put(d_content)
+     # <...>
```
Comment (daviddavis) [inline docs/plugins/plugin-writer/concepts/index.rst:73]: :+1:

File: docs/plugins/plugin-writer/concepts/index.rst (hunk 2 above)
Comment (daviddavis) [inline docs/plugins/plugin-writer/concepts/index.rst:58]:
```suggestion
In the stages pipeline, you want to intantiate the content units without saving them to database
```

**Pull request merged**